File: dtc.frontend/src/dtos/Round/CreateRoundDto.ts
import z from "zod";

export const createRoundDtoSchema = z.object({
  tournamentId: z.number({
    message: "TournamentId ist ein Pflichtfeld.",
  }),

  locationId: z.number({
    message: "LocationId ist ein Pflichtfeld.",
  }),

  sequence: z
    .number()
    .min(1, "Die Sequenz muss mindestens 1 sein.")
    .max(100, "Die Sequenz darf maximal 100 sein."),

  name: z
    .string()
    .max(50, "Der Rundenname darf maximal 50 Zeichen lang sein.")
    .nullable(),

  plannedStart: z.string().datetime({ offset: true }),

  plannedEnd: z.string().datetime({ offset: true }).nullable(),

  status: z
    .enum(["Scheduled", "InProgress", "Completed", "Cancelled"])
    .default("Scheduled"),
});

export type CreateRoundDto = z.infer<typeof createRoundDtoSchema>;

File: DTC.Api/Dtos/Round/CreateRoundDto.cs
using DTC.Api.Enums;
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Round
{
    public class CreateRoundDto
    {
        [Range(1, int.MaxValue, ErrorMessage = "Das Turnier ist ein Pflichtfeld.")]
        public int TournamentId { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Die Reihenfolge muss größer als 0 sein.")]
        public int Sequence { get; set; }

        [StringLength(100, ErrorMessage = "Der Name der Runde darf maximal 100 Zeichen lang sein.")]
        public string? Name { get; set; }

        [Required(ErrorMessage = "Der geplante Startzeitpunkt ist ein Pflichtfeld.")]
        public DateTimeOffset PlannedStart { get; set; }

        public DateTimeOffset? PlannedEnd { get; set; }

        public RoundStatus Status { get; set; } = RoundStatus.Scheduled;

        public RoundPhase Phase { get; set; } = RoundPhase.GroupStage;
    }
}
